### PR TITLE
Enable live tests for appconfiguration with native images

### PR DESCRIPTION
**Pull request by srnagar**
This PR is to set up native tests pipeline templates and use this pipeline to enable live testing with native images for App Configuration.

### CODE REVIEW

**Review by JimSuplizio — APPROVED** [review]
We went over things in a teams meeting. I'm good with these changes.

**Comment by benbp** [comment]
Sorry to throw a wrench in some of this as I know you've already had conversations and I may be missing some context, but this chnage strikes me as a lot of unnecessary yaml duplication that could be addressed with a couple template conditionals. Also the `native-tests.yml` file will not fit within our auto-pipeline generation conventions and it seems the pipelines were created manually. This will cause us maintainance problems in the future and the file should be called `tests.native.yml` instead.

**Comment by srnagar** [comment]
@benbp long term, the duplication will go away and existing live tests pipeline will be updated with conditionals to support this. However, native image support and live testing capabilities are still in early stages and modifying existing pipelines can block live tests from passing/releasing libraries if there are any issues with native image creation (which is known to happen as GraalVM frequently has breaking changes).

So, as discussed with @JimSuplizio  and @weshaggard, we went with creating a separate pipeline (with some duplication) until we are ready to integrate this with the main live tests pipeline.

Regarding auto pipeline generation, I can change `native-tests.yml` to `tests.native.yml` if that is the naming convention for pipeline generation.

File: eng/pipelines/templates/jobs/native.live.tests.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+      - ${{ if ne(parameters.DisableAzureResourceCreation, 'true') }}:
+          - template: /eng/common/TestResources/build-test-resource-config.yml
+            parameters:
+              SubscriptionConfiguration: ${{ parameters.CloudConfig.SubscriptionConfiguration }}
+              SubscriptionConfigurations: ${{ parameters.CloudConfig.SubscriptionConfigurations }}
+
+          - ${{ if parameters.TestResourceDirectories }}:
+              - ${{ each directory in parameters.TestResourceDirectories }}:
+                  - template: /eng/common/TestResources/deploy-test-resources.yml
```
[the remark this answers is not in the document]
Comment (srnagar) [reply eng/pipelines/templates/jobs/native.live.tests.yml:73]: This should be configured in `tests-resources.json` to ensure env vars of resources required for live tests aren't overwritten.

**Comment by benbp** [comment]
@srnagar ah I see, that makes sense thanks for explaining. Yes I would still do the yaml filename change.

**Pull request merged**